### PR TITLE
Sriharsha/reduction deployment

### PR DESCRIPTION
**Pull request by sriharsha219**
*Following changes/additions are made to support the leto reduction - cv2_jpg_reduction, ffmpeg_resolution_downsampler, and fps_bitrate modules deployment in target AWS EC2 instances using a common standard shell script which will be deployed in EC2 and triggered with a GitHub workflow.
*As of now we have the GitHub workflow created for the reduction modules deployment only and in the next PR, I am planning to develop some standard deployment workflow that can serve for all reconstruction modules deployment in the target AWS EC2 instance.
*This also contains a new workflow to run the leto - python scripts, which are already deployed in the EC2 instance.

Changes are done in the following files:
------------------------------------------
1.In the requiments*.txt files, updated the aEye version to 'aEye>=1.2.2'
2.Updated README.md file in '/deployment_scripts' folder, to correct the instructions as per the changes

New files:
----------
1./deployment_scripts/deploy-reduction-modules-in-ec2.sh
2..github/workflows/deploy-reduction-modules-in-ec2.yml
3..github/workflows/run-python-scripts-in-ec2.yml

Deleted files:
--------------
1.Deleted this module specific 'deployment_scripts/setup_and_run_ffmpeg_resolution_downsampler.sh' script as we have a new script that 
supports all modules deployment

### CODE REVIEW

**Review by hamzakhokhar — APPROVED** [review]
Everything looks good

File: deployment_scripts/README.md [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
-
-b)python3 opencv_resolution_upscaler.py
\ No newline at end of file
+Reduction modules deployment in AWS EC2 instance with shell scripts:
+--------------------------------------------------------------------
+The [reduction](https://github.com/DISHDevEx/leto/tree/main/reduction/) modules are used to take the input video file from the source S3 path, reduce the resolution, label it, and then upload the processed video to the destination S3 path.
+
+The module is deployed using shell scripts in AWS EC2 instance as per the below steps,
+
+1.Create an AWS EC2 Amazon Linux instance using 'Create new EC2 instance' workflow in the leto repository with the following parameters,
```
Comment (vinayakSharm) [inline deployment_scripts/README.md:7]: replace "parameters," with "parameters:"

File: deployment_scripts/README.md [hunk 2]
```diff
@@ -1,27 +1,46 @@
...
+
+a)Enter name of the EC2: "Enter desired name for EC2 instance"
+
+b)Enter AMI id for the EC2: ami-0f34c5ae932e6f0e4
```
Comment (vinayakSharm) [inline deployment_scripts/README.md:11]: b)Enter AMI id for the EC2: ami-0f34c5ae932e6f0e4 

change to 

b)Enter AMI id for the EC2:  {e.g./ ami-0f34c5ae932e6f0e4}

File: deployment_scripts/README.md [hunk 3]
```diff
@@ -1,27 +1,46 @@
...
+
+c)Enter EC2 instance type: t3.small
```
Comment (vinayakSharm) [inline deployment_scripts/README.md:13]: c)Enter EC2 instance type: {e.g. t3.small}

File: deployment_scripts/README.md [hunk 4]
```diff
@@ -1,27 +1,46 @@
...
+
+2.Login to the EC2 using the Session Manager to ensure it is up and running as expected. 
+  Also, copy the respective EC2 instance id as we have to pass it as a parameter in the next step.
+
+3.Then execute the 'deploy-reduction-modules-in-ec2' workflow in the leto repository with following parameters,
```
Comment (vinayakSharm) [inline deployment_scripts/README.md:18]: replace "parameters," with "parameters:"

File: deployment_scripts/README.md [hunk 5]
```diff
@@ -1,27 +1,46 @@
...
+
+  Note: This reduction module 'name' should be the same as the respective module 'folder name' under the 'reduction' directory in the GitHub - leto repository.
+
+4.The workflow will do the following steps in the EC2,
+
+a)Download the 'deploy-reduction-modules-in-ec2.sh' to EC2
+
+b)Execute the 'deploy-reduction-modules-in-ec2.sh' script in EC2
+
+5.The 'deploy-reduction-modules-in-ec2.sh' script will do the following steps in the EC2,
```
Comment (vinayakSharm) [inline deployment_scripts/README.md:34]: script will do the following steps in the EC2, 

which steps?

**Review by vinayakSharm — APPROVED** [review]
good.

**Pull request merged**